setup.py

```diff
@@ -11,19 +11,26 @@
 import os
 import sys
 from setuptools import setup
-
+from distutils import sysconfig
 
 long_description = """anyreadline tries to do the right thing and installs either 
 readline on MacOSX or pyreadline or Windows. 
-Or does nothing on Linux which comes in general with readline built-in"""
+On Linux and FreeBSD, it ensures readline to be built-in which is most of the
+case, otherwise the script fails."""
 
 sysplat = str(sys.platform).lower()
+has_readline = sysconfig.get_config_var('HAVE_LIBREADLINE')
 
 install_requires = ['setuptools'] 
 if 'darwin' in sysplat:
     install_requires += ['gnureadline']
 if 'win32' in sysplat:
     install_requires += ['pyreadline'] 
+elif 'linux' in sysplat or 'freebsd' in sysplat and has_readline:
```

Comment:
This would break every other OS. Like all the other BSDs. OS checks are inherently bad, unless we know the OS very well and it is known to be broken.

If any OS has libreadline, it is good.

If OS is Windows, use pyreadline.

If anything else (implicitly inc Darwin), add gnureadline as a dependency?
(Read discussion at #1 and at gnureadline issue.)